src/EventStore.Core/ClusterVNode.cs

```diff
@@ -508,7 +510,8 @@ protected virtual void OnNodeStatusChanged(VNodeStatusChangeArgs e)
                                                         !vNodeSettings.DisableScavengeMerging,
                                                         unsafeIgnoreHardDeletes: vNodeSettings.UnsafeIgnoreHardDeletes);
 
-			// ReSharper disable RedundantTypeArgumentsOfMethod
+            // ReSharper disable RedundantTypeArgumentsOfMethod
+            _mainBus.Subscribe<ClientMessage.MergeIndexes>(storageScavenger);
```

Comment:
It would be better if `ClientMessage.MergeIndexes` message is handled in `IndexCommitterService.cs` which is responsible for index-related stuff. The storage scavenger is responsible mainly for scavenging.